File: 3731-sum-of-variable-length-subarrays/sum-of-variable-length-subarrays.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums) {

        int n = nums.size();
        int ans = 0;
        for (int i = 0; i < n; i++) {
            int sum = 0;
            for (int j=max(0,i-nums[i]);j<=i;j++) {
                sum+=nums[j];
            }
            ans+=sum;
        }

        return ans;
    }
};

File: 3731-sum-of-variable-length-subarrays/sum-of-variable-length-subarrays.java
class Solution {
    public int subarraySum(int[] nums) {
        int sum=0;
        int n=nums.length;

        for(int i=0;i<n;i++){
            for(int j=Math.max(0,i-nums[i]);j<=i;j++){
                sum+=nums[j];
            }
        }

        return sum;
    }
}